timeseries.go

```diff
@@ -5,21 +5,36 @@ import (
 )
 
 // TimeSeries represents an array of candles
-type TimeSeries struct {
-	Candles []*Candle
+type TimeSeries interface {
+	LastIndex() int
+	FirstCandle() *Candle
+	LastCandle() *Candle
+	GetCandle(int) *Candle
+	GetCandleData() []*Candle
+	AddCandle(*Candle) bool
```

Comment:
in terms of the methods in this interface, I want to make sure we're being deliberate about what we choose, because once we commit to these we can't remove or change any of them. Based on the usages in this repo, I think we could get away with something like this:

```go
type ITimeSeries interface {
  AddCandle(*Candle) bool
  Candles() []*Candle
}
```
the others seem like convenience methods to access data at a particular location.

Reply:
Yeah, I understand your concerns. I guess that was a bit premature decision, disregarding legacy code. And presenting as small interface as possible is also a valid point. As for motivation behind this PR, it's just what I think is usually right: parts that "do" or "provide" stuff are usually interfaces, and parts that circulate around as DTOs are usually structs. The time series looks like a "provider" of Candles to me, hence it's an interface. But you're right, I should have thought it all through more carefully. For instance, some indicators use internal caches that also are slices. If we decide to go the whole way, cache must be more abstract too. For example, "infinite" time series implemented as some kind of ring buffer will not use too much memory, but cache as a slice definitely will.

Reply:
No worries! feel free to iterate on this and tag me for another review. Thanks for your contribution! It's exciting to see folks wanting to work on this and make it better